coral/src/services/api.ts

```diff
@@ -179,6 +179,17 @@ const checkStatus = (response: Response): Promise<Response> => {
   return Promise.resolve(response);
 };
 
+async function checkForKlawErrors<T>(response: Response): Promise<T | Error> {
+  const res = await response.json();
+  return new Promise((resolve, reject) => {
+    if (res.result && !res.result.toLowerCase().startsWith("success")) {
```

Comment:
Can you link the API type here? I think only checking a `result` from the response can result in unwanted behaviour. I would implement a super strict typeguard where we could check all other property names in the response and only then if the `result` is not "success"